tests/reporter/kubernetes.go

```diff
@@ -918,3 +1024,157 @@ func writeStringToFile(filePath string, data string) error {
 	_, err = f.WriteString(data)
 	return err
 }
+
+func getVmiType(vmi v12.VirtualMachineInstance) (string, error) {
+	for _, volume := range vmi.Spec.Volumes {
+		if volume.VolumeSource.ContainerDisk == nil {
+			continue
+		}
+
+		image := volume.VolumeSource.ContainerDisk.Image
+		if strings.Contains(image, "fedora") {
+			return "fedora", nil
+		} else if strings.Contains(image, "cirros") {
+			return "cirros", nil
+		} else if strings.Contains(image, "alpine") {
+			return "alpine", nil
+		}
+	}
+
+	return "", fmt.Errorf("unknown type, vmi %s", vmi.ObjectMeta.Name)
+}
+
+func prepareVmiConsole(vmi v12.VirtualMachineInstance, vmiType string) error {
+	err := console.SafeExpectBatch(&vmi, []expect.Batcher{
```

Comment:
the login fails if its already logged
for example if the test did login already and then asserted

Reply:
Our login methods should support re-login. If one of the three doesn't support it, it should be fixed.

Reply:
Done

Reply:
Note that the drawback of it (it was existing already on fedora and cirros, i just added to alpine)
is that it will consume several seconds in the tests themself for every first login